src/main/scala/util/ShiftReg.scala

```diff
@@ -84,11 +87,41 @@ object AsyncResetShiftReg {
     apply (in, depth, init.litValue.toInt, None)
 }
 
+class SynchronizerPrimitiveShiftReg(depth: Int = 1, init: Int = 0) extends AbstractPipelineReg(1) {
```

Comment:
I think if this is primarily intended as a synchronizer primitive the default depth should be 3, or we should just remove the default and require the user to specify it.

Reply:
Let's just build this with regular flops. And make sure it gets the proper inferred reset type.